### PR TITLE
Add Molex KK 41791 series

### LINKED ISSUE #423: Molex KK 396 is missing
Those connectors went missing: https://github.com/KiCad/Connectors_Molex.pretty/blob/master/Molex_KK-41791-02_02x3.96mm_Straight.kicad_mod. Is this intentional? I added them back then. Should I submit them again or should I host them somewhere else? Those connectors are typically used in pinball.

### PR DESCRIPTION
**Pull request by jabdoa2**
Fix #423

### CODE REVIEW

**Comment by poeschlr** [comment]
You would still need to open a footprint pull request on the official repo as this does not happen automatically (and the requests need to be linked with each other, otherwise it can easily happen that we miss the connection like i did with the connection between your issue and this pull request.)

File: scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        {'x': body_edge['left'], 'y': -sl/2},
+        {'x': body_edge['left'] + sl/sqrt(2), 'y': 0},
+        {'x': body_edge['left'], 'y': sl/2}
+    ]
+    kicad_mod.append(PolygoneLine(polygone=poly_pin1_marker, layer='F.Fab', width=fab_w))
+
+    yr1=body_edge['bottom']+nudge
+    yr2 = yr1 - 1
+    yr3 = yr2 - 0.53
+    if pincount <= 6:
```
Comment (poeschlr) [inline scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py]: The slot in the back plate is quite different in this series compared to the other KK series. It always has the same dimension but there are also parts with 2 slots. The position of where to find the slots is in the main table on page 1 of the datasheet.

File: scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    # calculate working values
+    end_pos_x = (pincount-1) * pitch
+    centre_x = (end_pos_x - start_pos_x) / 2.0
+    nudge = configuration['silk_fab_offset']
+    silk_w = configuration['silk_line_width']
+    fab_w = configuration['fab_line_width']
+
+
+    body_edge={
+        'left':start_pos_x - pitch/2,
```
Comment (poeschlr) [inline scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py]:
```suggestion
        'left':start_pos_x - 1.91,
```

File: scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        'right':end_pos_x + pitch/2,
```
Comment (poeschlr) [inline scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py]:
```suggestion
        'right':end_pos_x + 1.91,
```

File: scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        'bottom':5
```
Comment (poeschlr) [inline scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py]:
```suggestion
        'top':-5.11
```

File: scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        }
+    body_edge['top'] = body_edge['bottom']-10.01
```
Comment (poeschlr) [inline scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py]:
```suggestion
    body_edge['bottom'] = body_edge['top']+10.01
```

File: scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_ROUNDRECT
+
+    kicad_mod.append(PadArray(initial=1, start=[start_pos_x, 0],
+        x_spacing=pitch, pincount=pincount,
+        size=pad_size, drill=drill,
+        type=Pad.TYPE_THT, shape=Pad.SHAPE_OVAL, layers=Pad.LAYERS_THT,
+        **optional_pad_params))
+
+    # create fab outline
+    kicad_mod.append(RectLine(start=[body_edge['left'], body_edge['top']],\
```
Comment (poeschlr) [inline scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py]: The back plate does not extend till the end so a simple rectangle will not work here.

Comment (jabdoa2) [reply scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py]: Fixed in 84b21113c8de2851741bfadcff3f46d2ddd73319

File: scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py (hunk 1 above)
Comment (jabdoa2) [reply scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py]: Fixed in 84b21113c8de2851741bfadcff3f46d2ddd73319

**Comment by poeschlr** [comment]
thanks

**Pull request merged**